ansible/roles/nginx/defaults/main.yml

```diff
@@ -53,6 +53,31 @@ nginx_host_allow: []
 #
 nginx__deploy_state: 'present'
 
+                                                                   # ]]]
+# .. envvar:: nginx__packages_installation [[[
+#
+# There is a deployment scenario possible where you install nginx packages from
+# sources or manually. This option allows you to omit nginx packages
+# installation by the role. Use with caution, as it can have unexpected
+# consequences. Notice: you must define ``nginx_default_tls_protocols``
+# manually, for example:
+#
+# ``nginx_default_tls_protocols: [ "TLSv1.2", "TLSv1.3" ]``
+#
+# Possible options:
+#
+# ``True``
```

Comment:
Hi, I'm the doc guy who spoils everyone's fun, nothing personal :)

* "Unexpected consequences" is the opposite of documentation. Please make it clearer until nothing is unexpected.
* You don't need a whole boilerplate to document a boolean. I know `nginx__deploy_state` takes this approach, but check out `nginx_pki` for a shorter one.
* `warning` is used on `__deploy_state` because it **does not do the job it's expected to**. If your option does, explain its usage in regular paragraphs.
* Not a telegram. Full sentences. :P
* Oh is this the option for "expierenced" devops? lol when your doc is ready it will be for everyone.

> Boolean. Defauts to `True`. If set to `False`, the role will omit installation ...
   This is useful in cases where manual installation is desired, such as ...
   Note that `nginx_default_tls_protocols` will not be populated and...

Reply:
I can't forecast everything what can happen, as options are unlimited, depending on the way nginx binary is installed. I used that phrase to discourage people to use that switch, I really hope I achieved that reading your comments :+1:

Reply:
Wait, do you mean to **approve merging** that switch, or to just **use** it?

Reply:
I wrote ```use that switch``` above, I have not written anything about ```approval```. I have no idea how did you think that out, to be honest :)

Reply:
Okay, I'll try to be clear and very verbose. I might have confused you because I thought we were having a snarky, fun back-and-forth.

That red icon in the beginning of my comments is me requesting changes, before your PR can be merged into `debops`.

@drybjed helped you meet the code quality standards. Right now I'm volunteering my time to help you bring it up to the doc quality standards, so we can include your feature.

To address your design goal of "discouraging people to use the switch", no, it's not the way we do things. There are features much more complex than toggling a package installation, and we don't discourage people from using them. We **enable** them to.

Options are not unlimited. When the switch is `False` but the role still runs, it means `nginx` has somehow ended up on the machine, possibly after the `bootstrap` playbook has run but before the `site` playbook runs and errors out. This fact is not even mentioned in the documentation you provided.

As I mentioned in my original request for changes, the stuff for "experienced devops" users happens when developing the feature. When we're done, it will be a toggle like any other one, with clear instructions on how to implement it.
If you can't provide those instructions, then we don't know *anybody* who knows how to use the feature. If you don't even know how to use it, there's no reason to implement it.

If you need help implementing any of my requested changes, I'm here. If you have thoughts on better ways to do stuff than what I'm proposing, we can discuss those too, as we just did the subject of deterring people from using the toggle.

I'm going to bed now. The blockquote in my original message has an example of what the doc should look like to be consistent with the style of the rest of the project. Have a go and if you get stuck somewhere post here and I'll try to address it tomorrow.